Fix incorrect height of buttons placed in list

Mati365 opened the pull request:
Fix (theme-lark): Fixed incorrect (too short) height of buttons placed inside lists (e.g. buttons in annotations menus). This was an accidental regression introduced in one of the recent releases. Closed by #16409.

---

Closes cksource/ckeditor5-commercial#6248.

### Additional information

1. Regression introduced in this [commit](https://github.com/ckeditor/ckeditor5/commit/f072048026c5407467e343c116c71ff5a9d236b4#diff-89c661e74afa6f00cb58fc5bcd651f40c1750417127eae21b1d44d0f94ebf722L46-L50). It looks like accidental removal, menubar is not affected.
2. It fixes an issue causing incorrect height of heading dropdown items in our demos. See more in [comment](https://github.com/ckeditor/ckeditor5/pull/16409#issuecomment-2148971633). 
3. Further explanation https://cksource.slack.com/archives/C05R4RM7HRC/p1717416949627779

#### Before  
 

![image](https://github.com/ckeditor/ckeditor5/assets/3949262/e210a380-5bba-44e9-bf71-a11c5fec0622)
![obraz](https://github.com/ckeditor/ckeditor5/assets/3949262/8efe18d1-95b4-44ce-bfa2-1f308351bc35)

#### After 
 
![obraz](https://github.com/ckeditor/ckeditor5/assets/3949262/89ab33b6-8446-4501-8ae1-08fcb1363719)
![obraz](https://github.com/ckeditor/ckeditor5/assets/3949262/980321f1-4b56-4b03-8410-3d2fc9784d0d)

scofalik commented:
@oleq since you were responsible for the initial menu bar implementation, please verify if that was indeed accidental removal. If so, feel free to merge.

Also, it looks like a solid regression. I'd guess we affected multiple UI elements?

Mati365 commented:
@oleq / @scofalik I changed `line-height` to `calc(1 * var(--ck-line-height-base) * var(--ck-font-size-base))`, it looks fine. However, I spotted an issue in our [demo](https://ckeditor.com/ckeditor-5/demo/feature-rich/). It looks like `line-height` is not set in `Heading` dropdown and styling of parent container is affecting dropdown items height. 

![obraz](https://github.com/ckeditor/ckeditor5/assets/3949262/af56fbf7-c647-4d13-8577-7ab7fa89439a)

I added this [line](https://github.com/ckeditor/ckeditor5/pull/16409/files#diff-4344c9c7190eb04956ade79fc7c2ca8a5851d4f3dd571924e7846b8fc46e5785R23) to fix this.

oleq commented on packages/ckeditor5-heading/theme/heading.css:
```diff
@@ -18,3 +18,7 @@
 }
+
+.ck.ck-list__item > .ck-button[class*="ck-heading_heading"] .ck-button__label {
+	line-height: var(--ck-font-size-large);
```
I'm not so sure about this change. It creates a visual regression (see screenshots below)

**Current stable** https://ckeditor.com/docs/ckeditor5/latest/examples/builds/document-editor.html 

<img width="446" alt="image" src="https://github.com/ckeditor/ckeditor5/assets/1099479/6998d4ef-4af1-4be4-b595-b5120b8565a7">

**This PR**

<img width="783" alt="image" src="https://github.com/ckeditor/ckeditor5/assets/1099479/742ee650-f5ee-45f4-8b67-89ecf3b9f95d">

---

AFAIR this is customized on purpose by a dedicated CSS on our website

<img width="1919" alt="Screenshot 2024-06-19 at 14 49 33" src="https://github.com/ckeditor/ckeditor5/assets/1099479/1ca4a97f-71e1-4d12-afc6-4d2ee0ac0410">

Mati365 replied:
Yep, but on the other hand, it fixes overriding `line-height` CSS in new selector.  Without such rule it looks like this:

![obraz](https://github.com/ckeditor/ckeditor5/assets/3949262/0d430449-a367-428c-8d4b-c1624288e7bc)

Mati365 replied:
I managed to fix this in this [commit](https://github.com/ckeditor/ckeditor5/pull/16409/commits/f248f9f77fda933ef4abf5361e41297ff0d4780e#diff-89c661e74afa6f00cb58fc5bcd651f40c1750417127eae21b1d44d0f94ebf722R46). It should work properly.

oleq commented on packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/list/list.css:
```diff
@@ -43,6 +43,11 @@
 
+		& .ck-button__label {
+			/* https://github.com/ckeditor/ckeditor5-heading/issues/63 */
+			line-height: calc(1 * var(--ck-line-height-base) * var(--ck-font-size-base));
```
I think we can skip `1`.

oleq commented on packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/list/list.css:
```diff
@@ -43,6 +43,11 @@
 		   https://github.com/ckeditor/ckeditor5-heading/issues/63 */
 		padding: var(--ck-list-button-padding);
 
+		&:not([class*="ck-heading_heading"]) .ck-button__label {
```
I'm curious why we need this exception for heaings? Line heights in the heading dropdown menu have been consistent before I [introduced the regression](https://github.com/ckeditor/ckeditor5/commit/f072048026c5407467e343c116c71ff5a9d236b4#diff-89c661e74afa6f00cb58fc5bcd651f40c1750417127eae21b1d44d0f94ebf722L46) in 43.1.0. Am I missing something?

# 40.0.0

https://github.com/ckeditor/ckeditor5/assets/1099479/7d44470e-8593-459e-af30-0e9fbce83084

# 41.3.0 (regression introduced)

https://github.com/ckeditor/ckeditor5/assets/1099479/11f39646-5cb6-4d16-97a6-dcce77f1bb39

# This PR without exception

https://github.com/ckeditor/ckeditor5/assets/1099479/7d6b5045-3d50-4d8a-a484-aab0624559b9

Mati365 replied:
`line-height` is not set in `heading` plugin https://github.com/ckeditor/ckeditor5/blob/ck/6022/packages/ckeditor5-heading/theme/heading.css#L15 and setting it here can lead to more breaking changes in docs. From what I've seen, `line-height: auto` looks differently than absence of such attribute in these headers button and in terms of backward compatibility the lack of attribute is more safe.

Mati365 replied:
Without such exception, it looks like it:

![obraz](https://github.com/ckeditor/ckeditor5/assets/3949262/4ec941d8-1a5d-48b7-a4f4-e3dedf687dd1)

In our manual tests, which is not correct look, I assume.
